lib/db.js

```diff
@@ -916,7 +857,7 @@ module.exports = (
       .then(code => {
         const data = { uid, createdAt: Date.now(), flowId }
         return this.pool.put(`/signinCodes/${code.toString('hex')}`, data)
-          .then(() => code, err => {
+          .then(() => code.toString('base64'), err => {
```

Comment:
This change gives us a kind of weird division of responsibility between here and `lib/senders/sms.js`, which works in practice but includes an extra, unnecessary `toString('base64')`.

Would it be simpler / more self-contained if this method reverted to returning a `Buffer` so that the `toString('base64')` over there makes sense? Or if you don't like that option, would it make more sense if all of the code for converting to URL-safe base64 was pulled down to this level instead?

As it stands we have six of one and half a dozen of the other.